### PR TITLE
Hds 3383 - Update prettier and related packages

### PR DESCRIPTION
**Pull request by curthashicorp**
### Update to the latest version of the below plugins, in the following repos:
[ packages/components, packages/ember-flight-icons, showcase, website, codemods, packages/flight-icons ]

** While updating, we should strive to avoid opting in to a number of potentially disruptive changes.
```hds
"devDependencies": {
    "@typescript-eslint/eslint-plugin": "^6.21.0",
    "@typescript-eslint/parser": "^6.21.0",
    "ember-template-lint": "^6.0.0",
    "ember-template-lint-plugin-prettier": "^5.0.0",
    "eslint-config-prettier": "^9.1.0",
    "eslint-plugin-ember": "^11.12.0",
    "eslint-plugin-prettier": "^5.1.3",
    "ember-qunit": "^8.1.0",
    "prettier": "^3.3.2",
},
```

## Configuration Updates

The following rules were added to .eslintrc.js so that in our prettier config we can set trailingComma to es5 and avoid thousands of changes ( 'just to add a dangling comma to function parameters', ). My assumption is this value add wouldn't be worth the effort and would allow us to avoid inadvertently introducing a disruptive change.

```hds
  // Rules added to each .eslintrc.js
  rules: {
    'comma-dangle': [
      'error',
      {
        arrays: 'always-multiline',
        objects: 'always-multiline',
        imports: 'always-multiline',
        exports: 'always-multiline',
        functions: 'only-multiline',
      },
    ],
  },
```

```hds
// Add trailingComma rule to each .prettierrc.js
'use strict';
module.exports = {
    trailingComma: 'es5',
    overrides: [
```

```hds
// Added trailingComma to packages/flight-icons/scripts/build-parts/generateBundleSVGReact.ts
const prettierConfig = { parser: 'typescript' as const, tabWidth: 4, singleQuote: true, trailingComma: 'none' as const };
```

In regards to updating ember-template-lint and ember-template-lint-plugin-prettier I have added the following rules consistently across .template-lintrc.js
We might want to convert some of these new lint rules to 'warn' so that we are aware of code that we might want to refactor.  If we do not wish to do so we can leave them as is set to false. 

```hds
'use strict';

module.exports = {
  plugins: ['ember-template-lint-plugin-prettier'],
  extends: ['recommended', 'ember-template-lint-plugin-prettier:recommended'],
  rules: {
    'no-html-comments': false,
    'no-trailing-spaces': true,
    'no-negated-condition': false,
    'require-mandatory-role-attributes': false,
    'no-at-ember-render-modifiers': false,
    'no-unnecessary-curly-parens': false,
    'no-unnecessary-curly-strings': false,
    'no-redundant-role': false,
    'no-builtin-form-components': false,
  },
  ignore: ['tests/**'],
};
```

### Website
https://github.com/hashicorp/design-system/pull/2206/commits/ae2bfe4ae190422ceab4c16a1322be7e848bc010
https://github.com/hashicorp/design-system/pull/2206/commits/941bdcb462b70a44359651a420d1701f4b74c865
https://github.com/hashicorp/design-system/pull/2206/commits/9dfc7b71d0e7d3b1b6a14234cdfaf57a9cc0c58b

Small change to multi-line template literal syntax and prettier cleaned up some line endings:
website/app/components/doc/page/header/algolia-search/parts/htmlPanelNoResults.js
website/app/components/doc/page/header/algolia-search/parts/htmlPanelFooter.js

Add trailing commas previously missed in three files:
website/app/components/dynamic-template.js
website/app/initializers/showdown-extensions.js
website/app/routes/application.js

### Showcase
https://github.com/hashicorp/design-system/pull/2206/commits/1a64f3737908b8a5e51eb28091353b69f5b83ff7
https://github.com/hashicorp/design-system/pull/2206/commits/9dfc7b71d0e7d3b1b6a14234cdfaf57a9cc0c58b

only one file needed a few missing trailing commas:
showcase/types/global.d.ts

### packages/components
https://github.com/hashicorp/design-system/pull/2206/commits/4f8e9d4fcaa6236a1606e4b15ea182a2a1d8eabd
https://github.com/hashicorp/design-system/pull/2206/commits/9dfc7b71d0e7d3b1b6a14234cdfaf57a9cc0c58b

One or two lines changed in 17 files, mostly trailing commas and a couple removals of end of file returns
### packages/ember-flight-icons
https://github.com/hashicorp/design-system/pull/2206/commits/f266a76552752cdfaee8ac86fc200c7f4e5c3d58
https://github.com/hashicorp/design-system/pull/2206/commits/9dfc7b71d0e7d3b1b6a14234cdfaf57a9cc0c58b

No linting changes needed after update

### packages/flight-icons
https://github.com/hashicorp/design-system/pull/2206/commits/db82f0bd620f545b54bece18495460a43f2b728c
https://github.com/hashicorp/design-system/pull/2206/commits/de2fbc463504d8fb996923b42872f6472156e856

With the prettier update came Typescript updates. This method returns a  Promise<string>
packages/flight-icons/scripts/build-parts/getCssForIconAnimation.ts

Changes to these files were made to properly await async Promise from prettier.format calls
packages/flight-icons/scripts/build-parts/generateBundleSVG.ts
packages/flight-icons/scripts/build-parts/generateBundleSVGReact.ts

### packages/codemods
https://github.com/hashicorp/design-system/pull/2206/commits/ed9b19c729dd97ee95412a764a5a954d00e3d56c

A handful of small changes to have code in this repo adhere to prettier 3 defaults:
packages/codemods/transforms/v3/dropdown/index.js
packages/codemods/transforms/v3/dropdown/test.js
packages/codemods/transforms/v3/masked-input/index.js
packages/codemods/transforms/v3/masked-input/test.js
packages/codemods/transforms/v3/radio-card/test.js
packages/codemods/transforms/v3/side-nav/index.js
packages/codemods/transforms/v3/side-nav/test.js

### :link: External links

Jira ticket: [HDS-3383](https://hashicorp.atlassian.net/browse/HDS-3383)

[HDS-3383]: https://hashicorp.atlassian.net/browse/HDS-3383?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by curthashicorp** [comment]
need to fix some things

File: packages/codemods/package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "version": "0.2.0",
   "scripts": {
     "lint": "eslint --cache .",
+    "lint:fix": "eslint . --fix",
```
Comment (alex-ju) [inline packages/codemods/package.json:7]: 👍

**Review by alex-ju — APPROVED** [review]
Looking good! Always enjoying a clean git history 🧘

File: packages/flight-icons/scripts/build-parts/generateBundleSVG.ts [hunk 2]
```diff
@@ -47,9 +47,11 @@ export async function generateBundleSVG({ config, catalog } : { config: ConfigDa
         allIcons.push(`'${iconName}'`);
     }
 
-    // add CSS used to animate "loading" and "running" icons
-    await fs.writeFile(`${config.mainFolder}/svg/animation.css`, getCssForIconAnimation());
+    const animationIconCss:string = await getCssForIconAnimation();
```
Comment (didoo) [inline packages/flight-icons/scripts/build-parts/generateBundleSVG.ts:50]:
```suggestion
    const animationIconCss : string = await getCssForIconAnimation();
```

File: packages/flight-icons/scripts/build-parts/generateBundleSVGReact.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ import { ConfigData } from '../@types/ConfigData';
 import { AssetsCatalog } from '../@types/AssetsCatalog';
 import { getCssForIconAnimation } from './getCssForIconAnimation';
 
-const prettierConfig = { parser: 'typescript' as const, tabWidth: 4, singleQuote: true };
+const prettierConfig = { parser: 'typescript' as const, tabWidth: 4, singleQuote: true, trailingComma: 'none' as const };
```
Comment (didoo) [inline packages/flight-icons/scripts/build-parts/generateBundleSVGReact.ts:17]: should the `as const` go outside of the `{ ... }` block? 🤔

File: packages/flight-icons/scripts/build-parts/generateBundleSVGReact.ts [hunk 4]
```diff
@@ -124,5 +124,6 @@ export async function generateBundleSVGReact({ config, catalog } : { config: Con
     await fs.writeFile(`${config.mainFolder}/svg-react/types.ts`, typesContent);
 
     // add CSS used to animate "loading" and "running" icons
-    await fs.writeFile(`${config.mainFolder}/svg-react/animation.css`, getCssForIconAnimation());
+    const animationIconCss:string = await getCssForIconAnimation();
```
Comment (didoo) [inline packages/flight-icons/scripts/build-parts/generateBundleSVGReact.ts:127]:
```suggestion
    const animationIconCss : string = await getCssForIconAnimation();
```

**Review by didoo — COMMENTED** [review]
Left a few comments, but not blockers (I think)

File: packages/flight-icons/scripts/build-parts/generateBundleSVGReact.ts (hunk 3 above)
Comment (curthashicorp) [reply packages/flight-icons/scripts/build-parts/generateBundleSVGReact.ts]: Yes, good catch! Placing on outside will strongly type all options. Pushed update.

**Review by alex-ju — APPROVED** [review]
Thanks for the rebase – good to be merged!

**Pull request merged**